idseq_dag/util/convert.py

```diff
@@ -0,0 +1,77 @@
+import multiprocessing
+import random
+import subprocess
+import sys
+import threading
+import time
+from functools import wraps
+import idseq_dag.util.log as log
+
+def generate_unmapped_singles_from_sam(sam_file, output_fa):
```

Comment:
I think it would help simplify our codebase if we rewrote the “generate_unmapped” functions using samtools. We can get rid of some of the complexity

Reply:
E.g. see “samtools fasta” command with flag option here: http://www.htslib.org/doc/samtools.html

Reply:
We literally rewrote something that already exists

Reply:
samtools is also probably faster ;)

Reply:
I will add a todo for this. I think any external tool dependency needs to be well tested, which is not the goal for this migration.